src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs

```diff
@@ -17,60 +20,309 @@ namespace Microsoft.DotNet.Cli.NuGetPackageDownloader
+                    packageSourceLocation.NugetConfig.Value.GetDirectoryPath().Value;
+                string nugetConfigFileName = Path.GetFileName(packageSourceLocation.NugetConfig.Value.Value);
+                settings = Settings.LoadSpecificSettings(nugetConfigParentDirectory,
+                    nugetConfigFileName);
+            }
+            else
+            {
+                settings = Settings.LoadDefaultSettings(
+                    packageSourceLocation?.RootConfigDirectory?.Value ?? currentDirectory);
+            }
+
+            PackageSourceProvider packageSourceProvider = new PackageSourceProvider(settings);
+            defaultSources = packageSourceProvider.LoadPackageSources().Where(source => source.IsEnabled);
+
+            if (!packageSourceLocation?.OverrideSourceFeeds.Any() ?? true)
+            {
+                if (!defaultSources.Any())
+                {
+                    throw new NuGetPackageInstallerException("No NuGet sources are defined or enabled");
+                }
+
+                return defaultSources;
+            }
+
+            List<PackageSource> customSources = new List<PackageSource>();
+            foreach (string source in packageSourceLocation?.OverrideSourceFeeds)
+            {
+                if (string.IsNullOrWhiteSpace(source))
+                {
+                    continue;
+                }
+
+                PackageSource packageSource = new PackageSource(source);
+                if (packageSource.TrySourceAsUri == null)
+                {
+                    _logger.LogWarning(string.Format(
+                        LocalizableStrings.FailedToLoadNuGetSource,
+                        source));
+                    continue;
+                }
+
+                customSources.Add(packageSource);
+            }
+
+            IEnumerable<PackageSource> retrievedSources;
+            if (packageSourceLocation != null && packageSourceLocation.OverrideSourceFeeds.Any())
+            {
+                retrievedSources = customSources;
+            }
+            else
+            {
+                retrievedSources = defaultSources;
+            }
+
+            if (!retrievedSources.Any())
+            {
+                throw new NuGetPackageInstallerException("No NuGet sources are defined or enabled");
+            }
+
+            return retrievedSources;
+        }
+
+        private async Task<(PackageSource, IPackageSearchMetadata)> GetLatestVersionInternalAsync(
+            string packageIdentifier, IEnumerable<PackageSource> packageSources, bool includePreview,
+            CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(packageIdentifier))
+            {
+                throw new ArgumentException($"{nameof(packageIdentifier)} cannot be null or empty",
+                    nameof(packageIdentifier));
+            }
+
+            _ = packageSources ?? throw new ArgumentNullException(nameof(packageSources));
```

Comment:
Why this syntax here instead of following the same pattern as packageIdentifier?